src/inactive-blog-filter.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Returns whether or not the blog should be marked as inactive
+ * Criteria for filtering (in milliseconds) is based on filters.json under blog.inactive
+ */
+const fs = require('fs');
+const dotenv = require('dotenv');
+const feedParser = require('./feed-parser');
+
+// Configure the environment
+dotenv.config();
+
+/**
+ * Condition for passing redlist some() check
+ * @param {string} feedUrl - url of the feed to check against redlist
+ * @param {Object} redItem - object containing a redlist entry
+ * @returns {boolean} - bool value
+ */
+function isRedlisted(feedUrl, redItem) {
+  return (feedUrl === redItem.url);
+}
+
+/**
+ * Simple function that takes the difference between current date and post date
+ * If difference is greater than threshold, the blog is considered inactive
+ * @param {Date} postDate - javascript date object of the post being checked
+ * @returns {number} - date difference in milliseconds
+ */
+function dateDiff(postDate) {
+  const currentDate = new Date();
+  return currentDate - postDate;
+}
+
+/**
+* Callback for redlist check
+* @callback checkCallback
+* @param {boolean} result - true/false whether or not a feed url is redlisted
+*/
+
+/**
+ * Checks if feed url is redlisted
+ * @param {string} feedUrl - url of the feed to check against redlist
+ * @param {checkCallback} callback - a callback that runs after the check
+ */
+function check(feedUrl, callback) {
+  // Read redlist file
+  fs.readFile('feeds-redlist.json', (err, redListRaw) => {
+    if (err) {
+      // Error reading file
+      console.warn('problem reading redlist file:', err.message);
+      callback(false);
+      return;
+    }
+
+    if (redListRaw.length === 0) {
+      // File is empty
+      console.warn('redlist file is empty, no filters applied\n\tsee: feeds-redlist.json');
+      callback(false);
+      return;
+    }
+
+    // Concat to array no matter what
+    const redList = [].concat(JSON.parse(redListRaw));
+
+    callback(redList.some(isRedlisted.bind(null, feedUrl)));
+  });
+}
+
+/*
+ * Performs a separate sweep of all feeds to see which ones are inactive,
+ * then updates last post date in feeds-redlist.json
+ *
+ * Integration with system can be improved once feed-worker is augmented to
+ * pass feed data more intuitively
+ *
+ * Due to amount of operations, this can be run periodically instead of with every feed update
+ */
+function update() {
```

Comment:
The `update()` function is async, should we return a promise and only `resolve()` when it's done below?

Reply:
`update()` itself is not async, although it does use `fs.readfile`, but I chain directly off it with an anonymous callback. If we would like to make `update()` async, perhaps we could frame it as a separate issue?

Reply:
I believe you mention you addressed this in another push. If so, this should be resolved?

Reply:
@UltimaBGD Yes, I was waiting on the community to see if anyone had any inputs. Since it seems to be okay, I'll resolve this conversation and post an issue when this PR gets successfully merged.